internal/datasourcev2/polling_http_request.go

```diff
@@ -68,42 +71,110 @@ func (r *pollingRequester) BaseURI() string {
 func (r *pollingRequester) FilterKey() string {
 	return r.filterKey
 }
-func (r *pollingRequester) Request() ([]ldstoretypes.Collection, bool, error) {
+
+func (r *pollingRequester) Request() (*PollingResponse, error) {
 	if r.loggers.IsDebugEnabled() {
 		r.loggers.Debug("Polling LaunchDarkly for feature flag updates")
 	}
 
 	body, cached, err := r.makeRequest(endpoints.PollingRequestPath)
 	if err != nil {
-		return nil, false, err
+		return nil, err
 	}
 	if cached {
-		return nil, true, nil
+		return NewCachedPollingResponse(), nil
 	}
 
 	var payload pollingPayload
 	if err = json.Unmarshal(body, &payload); err != nil {
-		return nil, false, malformedJSONError{err}
+		return nil, malformedJSONError{err}
 	}
 
-	esEvents := make([]es.Event, 0, len(payload.Events))
-	for _, event := range payload.Events {
-		esEvents = append(esEvents, event)
+	parseItem := func(r jreader.Reader, kind datakinds.DataKindInternal) (ldstoretypes.ItemDescriptor, error) {
+		item, err := kind.DeserializeFromJSONReader(&r)
+		return item, err
```

Comment:
I believe this is equivalent:

```suggestion
		return kind.DeserializeFromJSONReader(&r)
```

Which makes `parseItem` a single line method. At that point, is there value in having it as a a separate function?